pkg/ebda/ebda.go

```diff
@@ -0,0 +1,120 @@
+// Copyright 2019 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package ebda looks for the EBDA pointer in /dev/mem, and provides access to the EBDA.
+// This is useful for us to read or write to the EBDA, for example to copy the RSDP
+// into the EBDA.
+package ebda
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+)
+
+const (
+	// EBDAAddressOffset is the traditional offset where the location of the EBDA is stored.
+	EBDAAddressOffset = 0x40E
+)
+
+// EBDA represents an EBDA region in memory
+type EBDA struct {
+	BaseOffset int64
+	Length     int64
+	Buf        []byte
+}
+
+func findEBDAOffset(f io.ReadSeeker) (int64, error) {
+	var ep uint16
+	if _, err := f.Seek(EBDAAddressOffset, io.SeekStart); err != nil {
+		return 0, fmt.Errorf("error seeking to memory offset 0x%X for the EBDA pointer, got: %v", EBDAAddressOffset, err)
+	}
+
+	// read the ebda pointer
+	if err := binary.Read(f, binary.LittleEndian, &ep); err != nil {
+		return 0, fmt.Errorf("unable to read EBDA Pointer: %v", err)
+	}
+
+	if ep == 0 {
+		return 0, errors.New("ebda offset is 0! unable to proceed")
+	}
+	// The EBDA pointer tends to be shifted 4 bits down.
```

Comment:
```suggestion
	// The EBDA pointer is shifted 4 bits left.
```

Reply:
No clue, probably some weird case.

Reply:
I thought it always was. 

Maybe it's not shifted 4 bits to the left when you are on a legit 16bit system?

Either way, I'd make clear in the comment that we think it's always.